pkg/server/svid/rotator.go

```diff
@@ -88,8 +89,14 @@ func (r *Rotator) rotateSVID(ctx context.Context) (err error) {
 		return err
 	}
 
+	serverID, err := idutil.ServerID(r.c.TrustDomain)
+	if err != nil {
+		// this should never fail; it is purely defensive
+		return fmt.Errorf("unable to determine server ID: %w", err)
```

Comment:
IMO, production code should ever purposefully panic. Also, if that happens our integration tests will catch it pretty early (since the server won't be able to obtain an SVID).